testnet/stacks-node/src/syncctl.rs

```diff
@@ -62,28 +62,17 @@ impl PoxSyncWatchdogComms {
         self.last_ibd.load(Ordering::SeqCst)
     }
 
-    /// Wait for at least one p2p state-machine passes
-    pub fn wait_for_p2p_state_pass(&self, timeout: u64) -> bool {
-        let current = self.get_p2p_state_passes();
-
-        let now = get_epoch_time_secs();
-        while current >= self.get_p2p_state_passes() {
-            if now + timeout < get_epoch_time_secs() {
-                debug!("PoX watchdog comms: timed out waiting for one p2p state-machine pass");
-                return false;
-            }
-            sleep_ms(1000);
-            std::sync::atomic::spin_loop_hint();
-        }
-        return true;
-    }
-
     /// Wait for at least one inv-sync state-machine passes
     pub fn wait_for_inv_sync_pass(&self, timeout: u64) -> Result<bool, burnchain_error> {
         let current = self.get_inv_sync_passes();
 
+        let now = get_epoch_time_secs();
         while current >= self.get_inv_sync_passes() {
```

Comment:
TL;DR We have to keep polling `self.get_inv_sync_passes()` or this loop will never exit.